### PR TITLE
CAL-55 Added integration test package and integration tests for NSILI

### PR DESCRIPTION
**Pull request by lcrosenbu**
#### What does this PR do?
Adding initial set of itests in Alliance, including NSILI Federated Source.
Separated the NSILI Mock Server and NSILI Client into two separate packages.

#### Who is reviewing it?
@peterhuffer @jrnorth @jaymcnallie @kcwire 

#### How should this be tested?
Build Alliance

#### What are the relevant tickets?
https://codice.atlassian.net/browse/CAL-55

#### Checklist:
- [X] Documentation Updated
- [ ] Update / Add Unit Tests
- [X] Update / Add Integration Tests

### CODE REVIEW

File: distribution/sdk/sample-nsili-client/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+-->
+<img src="https://tools.codice.org/wiki/download/attachments/1179800/ddf.jpg"/>
+# [Codice Alliance](http://github.com/codice/alliance/)
+
+## Mock NSILI Client
+
+Codice Alliance contains a sample NSILI client to be used for testing purposes.  These utilities can be run from the command-line within the sample-nsili-client directory by using maven and specifying a web port.
```
Comment (bdeining) [inline distribution/sdk/sample-nsili-client/README.md]: Nit : `This utility`

File: distribution/sdk/sample-nsili-server/README.md [hunk 2]
```diff
@@ -15,13 +15,8 @@
 
 ## Mock NSILI Server
 
-Codice Alliance contains a sample NSILI complaint server and client to be used for testing purposes.  These utilities can be run from the command-line within the sample-nsili-server directory by using maven and specifying a web and corba port.
+Codice Alliance contains a sample NSILI compliant server to be used for testing purposes.  These utilities can be run from the command-line within the sample-nsili-server directory by using maven and specifying a web and corba port.
```
Comment (bdeining) [inline distribution/sdk/sample-nsili-server/README.md]: Nit : `This utility`

File: distribution/sdk/sample-nsili-client/README.md (hunk 1 above)
Comment (lcrosenbu) [inline distribution/sdk/sample-nsili-client/README.md]: Resolved.

**Comment by lcrosenbu** [comment]
Resolved.

File: distribution/sdk/sample-nsili-server/README.md (hunk 2 above)
Comment (lcrosenbu) [inline distribution/sdk/sample-nsili-server/README.md]: Resolved.

**Comment by peterhuffer** [comment]
Built against an empty repository and got the following test failures
`Failed tests: 
  testNsiliSourceOpenSearchGetAll(alliance.test.itests.TestFederation): Failed to setup exam for TestFederation: The bundle Alliance :: NSILI :: Endpoint failed.
  testNsiliSourceOpenSearchLocation(alliance.test.itests.TestFederation): Exam setup failed
  testNsiliSourceAvailable(alliance.test.itests.TestFederation): Exam setup failed`

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/Nsili.java [hunk 3]
```diff
@@ -24,6 +25,7 @@
 public interface Nsili {
 
     @GET
-    @Produces(MediaType.WILDCARD)
+    @Consumes({MediaType.TEXT_PLAIN, MediaType.APPLICATION_XML, MediaType.APPLICATION_OCTET_STREAM, MediaType.WILDCARD})
```
Comment (jaymcnallie) [inline catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/Nsili.java]: If we have to explicitly add all these, what is `MediaType.WILDCARD` doing exactly?

File: distribution/test/itests/pom.xml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                        <goals>
+                            <goal>generate-depends-file</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+        <pluginManagement>
+            <plugins>
+                <!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on the Maven build itself.-->
```
Comment (jaymcnallie) [inline distribution/test/itests/pom.xml]: Why are we storing this?

**Comment by jaymcnallie** [comment]
Test setup is failing b/c it's still looking for features that were removed with https://github.com/codice/alliance/pull/61.  Rebase and the container should start up.

**Comment by lcrosenbu** [comment]
Updated to resolve issue reported by @peterhuffer. Default ORB port changed to deconflict with the port allocations done by the itests (see AbstractIntegrationTest).

File: distribution/sdk/sample-nsili-client/pom.xml [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        <artifactId>sdk</artifactId>
+        <version>0.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <name>Alliance :: Distribution :: SDK :: NSILI Client</name>
+    <artifactId>sample-nsili-client</artifactId>
+    <packaging>bundle</packaging>
+
+    <properties>
+        <checkstyle.skip>true</checkstyle.skip>
```
Comment (peterhuffer) [inline distribution/sdk/sample-nsili-client/pom.xml]: Is this being used anywhere?

**Comment by peterhuffer** [comment]
Successfully built a full build against an empty repository. LGTM

File: distribution/sdk/sample-nsili-client/pom.xml (hunk 5 above)
Comment (lcrosenbu) [inline distribution/sdk/sample-nsili-client/pom.xml]: The NSILI Client is a developer testing tool that can be launched from the command line. Refer to the README under this component for usage.

Comment (lcrosenbu) [inline distribution/sdk/sample-nsili-client/pom.xml]: This was part of refactoring the NSILI Client into it's own component sample-nsili-client. The `checkstyle.skip` property appears to have been used because this command-line test tool contains System.out statements.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java [hunk 6]
```diff
@@ -46,7 +46,7 @@
 
      public static final int DEFAULT_MAX_NUM_RESULTS = 500;
 
-    public static final int DEFAULT_CORBA_PORT = 20003;
+    public static final int DEFAULT_CORBA_PORT = 2809;
```
Comment (kcwire) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java]: Why the port change?

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/Nsili.java [hunk 7]
```diff
@@ -24,6 +25,7 @@
 public interface Nsili {
 
     @GET
-    @Produces(MediaType.WILDCARD)
+    @Consumes({MediaType.TEXT_PLAIN, MediaType.APPLICATION_OCTET_STREAM, MediaType.WILDCARD})
```
Comment (kcwire) [inline catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/Nsili.java]: This method doesn't appear to actually consume anything - why was this added?

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java (hunk 6 above)
Comment (lcrosenbu) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java]: Default ORB port 20003 changed to deconflict with the port allocations done by the integration tests (see AbstractIntegrationTest). This conflict caused an issue where the NSILI endpoint could not initialize when running itests. We investigated a standard port for IIOP and did not find one universally used.

File: catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/Nsili.java (hunk 7 above)
Comment (lcrosenbu) [inline catalog/nsili/catalog-nsili-common/src/main/java/org/codice/alliance/nsili/common/Nsili.java]: These changes were made to sync with another debugging effort that is not yet committed, and I pulled them in to ensure they were working for my test case.

File: catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java (hunk 6 above)
Comment (lcrosenbu) [inline catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java]: Default ORB port 20003 changed to deconflict with the port allocations done by the integration tests (see AbstractIntegrationTest). This conflict caused an issue where the NSILI endpoint could not initialize when running itests. We investigated a standard port for IIOP and did not find one universally used.

File: distribution/test/itests/pom.xml (hunk 4 above)
Comment (lcrosenbu) [inline distribution/test/itests/pom.xml]: Reused from DDF.

File: distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+    @Override
+    protected Option[] configureCustom() {
+        return options(
+                wrappedBundle(mavenBundle("ddf.test.itests", "test-itests-ddf").classifier("tests")
+                        .versionAsInProject()).bundleSymbolicName("test-itests-ddf")
+                                .exports("ddf.test.itests.*"),
+                wrappedBundle(mavenBundle().groupId("org.codice.alliance.distribution")
+                        .artifactId("sample-nsili-server").versionAsInProject()),
+                wrappedBundle(mavenBundle().groupId("ddf.catalog.core")
```
Comment (kcwire) [inline distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java]: Wrapping bundles can be dangerous and avoided.  This will expose the classes within the jar and could potentially mask a deployment issue. I think we need to remove these and get the tests running with only the ddf-itests & sample server jar.

File: distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+import ddf.common.test.BeforeExam;
+import ddf.test.itests.AbstractIntegrationTest;
+
+
+/**
+ * Tests the Alliance additions to DDF framework components.
+ */
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class TestFederation extends AbstractIntegrationTest {
```
Comment (kcwire) [inline distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java]: Should this be called `TestNsili` ?

File: distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+
+        MockNsiliRunnable mockServer = new MockNsiliRunnable(Integer.parseInt(webPort.getPort()),
+                Integer.parseInt(corbaPort.getPort()));
+        mockServerThread = new Thread(mockServer, "mockServer");
+        mockServerThread.start();
+    }
+
+    private void configureNsiliSource() throws IOException {
+        NsiliSourceProperties sourceProperties = new NsiliSourceProperties(NSILI_SOURCE_ID,
+                "file://" + System.getProperty("user.dir") + "/../../ior.txt");
```
Comment (kcwire) [inline distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java]: relative paths are bad - is this just in the test resources?

File: distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java (hunk 9 above)
Comment (lcrosenbu) [inline distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java]: Resolved.

File: distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java (hunk 10 above)
Comment (lcrosenbu) [inline distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java]: Resolved. Refactored to remove use of IOR File, and serve IOR content via MockWebService. Relative paths are OBE.

**Comment by kcwire** [comment]
👍

File: distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java (hunk 8 above)
Comment (lcrosenbu) [inline distribution/test/itests/test-itests-alliance/src/test/java/alliance/test/itests/TestFederation.java]: Resolved. Refactored wrappedBundle dependencies as <Embed-Dependency> in the pom.xml.

**Comment by jaymcnallie** [comment]
Pulled latest and built on a clean repo.  All itests passed.  👍
